### PR TITLE
[Fix #7144] Extend the namespaces definition for Style/Documentation cop

### LINKED ISSUE #7144: Style/Documentation accept constant visibility declaration as namespace
When declare the visibility of a constant (class and module include), the namespace object get an offense to document it.

```
module Namespace
  class Private
  end

  private_constant :Private
end
```

--------

## Expected behavior

Don't get an offense on the Namespace module. by considering constant visibility declaration as constant declaration.

## Actual behavior

Get an offense on the Namespace module.

## Steps to reproduce the problem

* Create a file with the code above
* Run Rubocop on it

## RuboCop version

```
0.71.0 (using Parser 2.6.3.0, running on ruby 2.4.5 x86_64-darwin18)
```

### PR DESCRIPTION
**Pull request by AdrienSldy**
Extend the definition of namespaces by adding the constants visibility
statement.
No longer report an error for classes and modules that contain constants'
visibility statements.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/documentation.rb [hunk 1]
```diff
@@ -3,10 +3,11 @@
 module RuboCop
   module Cop
     module Style
-      # This cop checks for missing top-level documentation of
-      # classes and modules. Classes with no body are exempt from the
-      # check and so are namespace modules - modules that have nothing in
-      # their bodies except classes, other modules, or constant definitions.
+      # This cop checks for missing top-level documentation of classes and
```
Comment (bbatsov) [inline lib/rubocop/cop/style/documentation.rb]: I wonder if we shouldn't add here examples of the code that's exempt, so it's clearer for users what the cop acts on and what it ignores.

**Comment by bbatsov** [comment]
You're suggestions makes sense to me, but I think we should extend the documentation a bit to make it clearer what exactly are we ignoring.

File: lib/rubocop/cop/style/documentation.rb [hunk 2]
```diff
@@ -31,6 +32,9 @@ class Documentation < Cop
 
         def_node_matcher :constant_definition?, '{class module casgn}'
         def_node_search :outer_module, '(const (const nil? _) _)'
+        def_node_matcher :constant_visibility_declaration?, <<-PATTERN
```
Comment (koic) [inline lib/rubocop/cop/style/documentation.rb]: Can you rebase with latest master branch and use squiggly heredoc?

```diff
- def_node_matcher :constant_visibility_declaration?, <<-PATTERN
+ def_node_matcher :constant_visibility_declaration?, <<~PATTERN
```

Comment (AdrienSldy) [reply lib/rubocop/cop/style/documentation.rb]: @koic done :)

File: lib/rubocop/cop/style/documentation.rb (hunk 1 above)
Comment (AdrienSldy) [reply lib/rubocop/cop/style/documentation.rb]: Ok, I think you're right. I will do it soon.

Comment (AdrienSldy) [reply lib/rubocop/cop/style/documentation.rb]: @bbatsov Sorry for the time spent. What do you think of comments? That's enough for you as an explanation?

File: lib/rubocop/cop/style/documentation.rb [hunk 3]
```diff
@@ -18,19 +19,52 @@ module Style
...
+      #   module Math
+      #   end
+      #
       #   # good
       #   # Description/Explanation of Person class
       #   class Person
       #     # ...
       #   end
       #
+      #   # allowed
```
Comment (bbatsov) [inline lib/rubocop/cop/style/documentation.rb]: allowed -> good

File: lib/rubocop/cop/style/documentation.rb [hunk 4]
```diff
@@ -18,19 +19,52 @@ module Style
...
+      #     # Class without body
```
Comment (bbatsov) [inline lib/rubocop/cop/style/documentation.rb]: You should indent the new examples just as much as the existing ones.

**Comment by bbatsov** [comment]
The changes look good, but your branch has to be rebased on top of the current `master` branch due to merge conflicts.

**Pull request merged**